reference/generators/custom.rst

```diff
@@ -0,0 +1,91 @@
+.. _custom_generator:
```

Comment:
I wouldn't add this to the list of _available generators_... I would put a comment below that list ("if none of these generators fits your needs, you can write a custom generator"):

![image](https://user-images.githubusercontent.com/1406456/56949426-2d5bb600-6b33-11e9-986d-61976fe47d42.png)

linking to the `howtos/custom_generators` or `integrations/other` and merge all this content there, otherwise we will have three sections of the documentation talking about the same: how to write a custom generator, and it is too much from my POV.

Reply:
I understand what you mean, but we have different possibilities. The section at ``reference/generators/custom.rst`` is intended to be a reference of the generator class and the available attributes and properties. So I don't like the idea of moving it out of the reference but agree that making it part of the list is probably not ideal.

The integration part is a more generic one talking about the json generator, txt one, deps_cpp_info...

An finally the implementation of a generator package is covered in a how-to. All sections are properly cross-referenced and I think it makes